scrapy/core/downloader/handlers/s3.py

```diff
@@ -24,21 +24,27 @@ def __init__(self, settings, *,
         # If no credentials could be found anywhere,
         # consider this an anonymous connection request by default;
         # unless 'anon' was set explicitly (True/False).
-        anon = kw.get('anon')
-        if anon is None and not aws_access_key_id and not aws_secret_access_key:
-            kw['anon'] = True
-        self.anon = kw.get('anon')
+        self.anon = kw.pop('anon', None)
+        if kw:
+            raise TypeError(f'Unexpected keyword arguments: {kw}')
 
         self._signer = None
         import botocore.auth
-        import botocore.credentials
-        kw.pop('anon', None)
-        if kw:
-            raise TypeError(f'Unexpected keyword arguments: {kw}')
-        if not self.anon:
-            SignerCls = botocore.auth.AUTH_TYPE_MAPS['s3']
-            self._signer = SignerCls(botocore.credentials.Credentials(
-                aws_access_key_id, aws_secret_access_key))
+        import botocore.session
+        if self.anon is None or self.anon is False:
+            session = botocore.session.get_session()
+            # Trigger credentials creation with Session.create_client().
+            session.create_client(
```

Comment:
Firstly I intended to only extract [credentials related code in `Session.create_client()`](https://github.com/boto/botocore/blob/c329fd23d3672b724a6dbd152f738de0b4d5a7ac/botocore/session.py#L825-L839). 

But using `Session.create_client()` here makes the code cleaner. Besides, considering we have already been using `Session.create_client()` in pipelines and feed exporter. Sticking to the same method in `S3DownloadHandler` maximize consistence across the project.